### DISCUSSION TITLE
Idea: Multiple Far Windows

### DISCUSSION THREAD

**Discussion by jacquesvandermerwe** [rxhanson/Charmstone-Community#3] [Ideas]
Love CharmStone and bought the app, but with all my scripts I ran out of slots on the far window.  😄 

Would it be possible to create multiple far windows in North, East, South, West position, so that when we move the mouse into those quadrants it shows that particular far configuration.

Thanks

**Comment by rxhanson** [comment]
Sorry for the delayed response here. Running out of slots appears to be a common issue. The solution that I've been thinking of is a notion of "Stacks" where you can have multiple items in one slot, maybe something that can pop out on hover or click, maybe something where the Charmstone view disappears and just the stack view is displayed. I'll have to think on this one a little more, and see if I can make something elegant.

**Comment by jacquesvandermerwe** [comment]
I think that in combination with Folders which can act as categories e.g. a Folder called 'Scripts' that then pops out as you said to show the scripts that can be run. Sounds good.

**Comment by SENTINELITE** [comment]
Wanted to add to this. 

I think having "profiles" makes sense. For example, you can create another profile, which you can swap to with ⌘+1-9. Perhaps you could also use the scroll wheel when centered on Charmstone's interface to scroll through profiles. 

I imagine this would work really well if you use multiple workflows. Ie, 
- Dev (PyCharm, Terminal, Xcode, Browser, VSCode, etc.)
- Art (Photoshop, Pixelmator, XD, Figma, etc.)
- Editing (Final Cut, Photoshop, Descript, etc.)

The stacks idea seems cool, but from my thinking, I don't think this would feel very "Charmstone-y". 😄

**Reply by rxhanson** [reply]
Thanks for the feedback! It's been a while since a Charmstone update, so hopefully I can make it back for another round of new features on it soon. 

I always feel torn when thinking through the "right" thing to do on this one. I really like the profiles idea, but I'm not quite sold on it yet. Configuration turns into a list view of profiles, and that adds a degree of complexity with slightly less approachability than what's currently there. There is potential that there is no way around that degree of added complexity in order to solve this problem, I just want to make sure I get it right if that's the case.

**Reply by CharlieDixon** [reply]
@SENTINELITE I love this idea of profiles, seeing as many people juggle different hats with a set of tools for each

**Reply by rxhanson** [reply]
I've given this one some more thought, and I think I have an idea in mind that I like. Effectively, borrow some UI design from the macOS launchpad, or iOS app groups, where I would have the little white dots on the bottom that denote which "page" you're on, and you can add multiple pages using scrolling to "swipe" between them. The "page" can be remembered per display. Configuring pages would have some tedium in the UI, unfortunately, and would probably take the form of tabs that can be added. In my backlog, I have a feature for easier configuration using drag and drop into the Charmstone window as opposed to the prefs window, and it might make sense to implement that first. At any rate, it might take me a while to get this one out the door as I have numerous items on other apps in line first.

**Reply by rn1d** [reply]
I like the idea of profiles. 

@rxhanson, I came here to ask for a similar feature. I would also like to have different sets of apps for different contexts, i.e. work and personal. 

My idea was to use Focus Filters for automatic switching. There could be multiple sets of apps (profiles) and the user would associate it with one (or possibly more) Focus modes. When the Focus mode is enabled, Charmstone would automatically switch to the mapped profile, if a mapping exists. Please consider that as well. 

For what it's worth, configuring the list via preferences window is completely fine for me. Drag and drop in the Charmstone switcher window directly isn't something I particularly want.

Thank you for the app, it works really well for my needs! Happy to have paid for the full version unlock.

**Reply by aaronmw** [reply]
@rxhanson I like the idea of "sheets" of icon grids. "Profiles" seem sort of extreme. I wonder just how many apps people are realistically switching between on such a regular basis anyway.

One "sheet" for messenging apps, another for work apps, another for whatever else... I could probably add a few more apps to my switcher if I had more than the 16 slots available, but I'd struggle. 

I (biased) still think the idea of a flexible-sized circular layout is the most elegant way of affording more icons. It eliminates the need to introduce any new complexity, and even obsoletes the current model of near / far, which actually requires I maintain two identically-shaped mental maps of my apps, whereas the circle ensures each app has its own position on the "clock".

**Reply by rxhanson** [reply]
The benefit of going with profiles is that there's more flexibility, with the downside of more complex configuration. For example, you could have Charmstone auto-switch between different selections based on a number of things, like which display you're on, which app is frontmost, your focus mode, your location (work/home), etc. I can see this being beneficial, but one of the things I've always liked about Charmstone is that it's pretty simple as-is, and adding in more complexity for something like this might only benefit a small percentage of users. (It's hard to pull features out once they're in, so I've learned that you really have to want the added complexity). 

I view the circular layout in sort of a separate realm from the profiles. 

Both of these would require some overhaul to the configuration UI, and effort to reduce the complexity of adding them in while retaining the simplicity of what's currently there.

**Reply by vwal** [reply]
I'd vote for an optional additional "spheres" (rungs? stratums? 🙂)  for near and far, OR a stack-ish approach where a hovered item could fan out to reveal additional items. I feel that sheets/leafs/pages would not lend themselves as easily to the extremely fluid way Charmstone works currently. Also, in my use case, I don't have completely fenced-off use cases, but just quite a few tools I use throughout the day. The tools are usually spread/arranged across multiple Spaces, and I use Charmstone to switch between them.
